packages/cli-kit/src/git.test.ts

```diff
@@ -50,37 +64,165 @@ describe('downloadRepository()', () => {
+  })
+})
+
+describe('commitAll()', () => {
+  it('throws if no files to commit', async () => {
+    mockedStatus.mockResolvedValue({files: []})
+
+    await expect(git.commitAll('')).rejects.toThrowError(git.NothingToCommitError())
+  })
+  it('tracks all files and commit', async () => {
+    mockedStatus.mockResolvedValue({files: [{path: '/some/path'}]})
+    mockedCommit.mockResolvedValue({commit: 'sha'})
+    const commitMsg = 'my msg'
+
+    const commitSha = await git.commitAll(commitMsg)
+
+    expect(mockedStatus).toHaveBeenCalledTimes(1)
+    expect(mockedAdd).toHaveBeenCalledTimes(1)
+    expect(mockedAdd).toHaveBeenCalledWith(['/some/path'])
+    expect(mockedCommit).toHaveBeenCalledTimes(1)
+    expect(mockedCommit).toHaveBeenCalledWith(commitMsg, undefined)
+    expect(commitSha).toBe('sha')
+  })
+  it('passes options to relevant function', async () => {
+    const author = 'Vincent Lynch <vincent.lynch@shopify.com>'
+    const directory = '/some/path'
+    mockedStatus.mockResolvedValue({files: [{path: '/some/path'}]})
+    mockedCommit.mockResolvedValue({commit: 'sha'})
+
+    await git.commitAll('msg', {author, directory})
+
+    expect(simpleGit).toHaveBeenCalledWith({baseDir: directory})
+    expect(mockedCommit).toHaveBeenCalledWith('msg', {'--author': author})
+  })
+})
+
+describe('getHeadSymbolicRef()', () => {
+  it('gets git HEAD symbolic reference', async () => {
+    const testRef = 'refs/heads/my-test-branch'
+
+    mockedRaw.mockResolvedValue(testRef)
+
+    await expect(git.getHeadSymbolicRef()).resolves.toBe(testRef)
+  })
+  it('throws if HEAD is detached', async () => {
+    const testRef = ''
+
+    mockedRaw.mockResolvedValue(testRef)
+
+    await expect(git.getHeadSymbolicRef()).rejects.toThrowError(git.DetachedHeadError())
+  })
+  it('passes the directory option to simple git', async () => {
+    // Given
+    const directory = '/test/directory'
+    mockedRaw.mockResolvedValue('ref/unit')
+
+    // When
+    await git.getHeadSymbolicRef(directory)
+
+    // Then
+    expect(simpleGit).toHaveBeenCalledWith({baseDir: directory})
+  })
+})
+
+describe('ensurePresentOrAbort()', () => {
+  it('throws an error if git is not present', async () => {
     // Given
     vi.mocked(hasGit).mockResolvedValue(false)
 
     // Then
-    await expect(() => ensurePresentOrAbort()).rejects.toThrowError(GitNotPresentError())
+    await expect(git.ensurePresentOrAbort()).rejects.toThrowError(git.GitNotPresentError())
   })
 
-  test("doesn't throw an error if Git is present", async () => {
+  it("doesn't throw an error if Git is present", async () => {
     // Given
     vi.mocked(hasGit).mockResolvedValue(true)
 
     // Then
-    await expect(ensurePresentOrAbort()).resolves.toBeUndefined()
+    await expect(git.ensurePresentOrAbort()).resolves.toBeUndefined()
+  })
+})
+
+describe('ensureInsideGitDirectory()', () => {
+  it('throws an error if not inside a git directory', async () => {
```

Comment:
I'd recommend doing `expect(async () => { await subjectFunction() })` so that the asynchronous function logic is executed by Vitest when the test runs and not when the test is declared. This comment applies to all the changes in this file.